### DISCUSSION TITLE
Additional FFMPEG code options

### DISCUSSION THREAD

**Discussion by age3rcm** [opentoonz/opentoonz#6573] [Ideas]
### Is your feature request related to a problem? Please describe

better management of computer resources

### Describe the solution you'd like

Possibility to add own code for FFMPEG export options like -HW acceleration

**Comment by age3rcm** [comment]
renember, using GPU to render maybe not produce the best results. but to use it as a preview of your product to be quikly rendered have advantages over CPU only and also having produced video file to send a preview to your client

**Comment by RodneyBaker** [comment]
I made a little progress on this front by creating a command line console program (C++) that takes its ffmpeg instructions from an external file.  I need to learn how to move such console programs over to Qt but so far so good.

The instructions might be better suited to be in XML or JSON format but currently I just have standard command line entries:

> ffmpeg -i input.png output.ppm
> ffmpeg -i output.ppm output.0001.png
> //This is a commented line that will not run ffmpeg -i output.0001.png comment.jpg
> ffmpeg -i output.0001.png output.0002.png
> END  This is where the instructions will end
> ffmpeg -i output.0001.png notread.png <as it is after the END command this line should not run

In theory this could run instructions with other command line programs but I'm not sure that would be the most secure approach.

Also, my utility is  WIndows only.

What I envision is more of a table where entries can be turned on/off but also reordered.
The data would likely be stored in an external file so that it could be edited outside of Opentoonz.

**Comment by age3rcm** [comment]
this is wonderful!
will a median user with no experience on code be able to use it tho?

**Comment by RodneyBaker** [comment]
It's more research than anything else.  While I might put the code (and program) out on my github I wouldn't really expect anyone to use it.  What a median user will use... or any Opentoonz user for that matter... use is what the research is for.
That eventual feature that I have in my imagination.

Short answer:  no.  :)

**Comment by RodneyBaker** [comment]
As this is likely inevitable and mostly a matter of how we want to proceed, will transfer this to Discussions.
